docs/design/reserve-action.md

```diff
@@ -0,0 +1,261 @@
+
+There are many reasons why a task can not be scheduled. From the perspective of task resource requirement and cluster 
+allocatable resources, if the resources required by the task is larger than the allocatable resource of any node in the 
+cluster, the task can not be scheduled. The node under the cluster needs to be expanded vertically to increase the node 
+resources. The `reserve` action does not solve scheduling problem of tasks that cannot be scheduled because the required 
+resource is larger than the allocated resources of any node in cluster.
+
+If the allocated node resources in the cluster can meet the resource requirements of the task, but because of that some of the 
+node resources are occupied by pods with relatively small resource requirements for a long time, the remaining resources of 
+the node can not satisfy the scheduling condition of the task with larger resource requirement, these tasks that apply for 
+large resources can not be scheduled due to insufficient resources. In extreme cases, tasks with large resource requirement 
+can only be scheduled when all the small resource application pods are completed. This situation needs to be avoided as much
+as possible. When a task cannot be scheduled for a long time, a special scheduling mechanism need to be provided for it to
+ensure that the task can be scheduled as soon as possible.
+
+![pod-starving](../../docs/images/pod-starving.png)
+
+In `Time0`, there are seven pods to be scheduled with one node whose allocated resources are `3C3G` in cluster. The `pod-0` has
+the highest priority in the unscheduled pod queue and will be prioritized to be schedule. In `Time1`, `pod-1` `pod-2` `pod-3` 
+were bound to the node and begin to run. The `pod-4` has the highest priority in the unscheduled pod queue and will be 
+prioritized to be schedule. In `Time2`, `pod-1` was completed, and `1C1G` resource was released from node. The scheduler
+try to schedule `pod-4`, but idle resource of node can not satisfy pod's requirement. The `pod-4` can not be scheduled and 
+was skipped. The schedule try to schedule `pod-5` and succeed in `Time3`. In `Time4`, `pod-2` and `pod-3` were `completed`
+and `2C2G` was released from node. Because of that the idle resource of node can not satisfy requirement of pod `pod-4`, `pod-4`
+was skipped. The `pod-6` and `pod-7` were bound to node and begin to run. In `Time6` all pods except pod `pod-4` were completed
+and all resources of node were release. The scheduling condition of pod `pod-4` is satisfied, `pod-4` was bound to node and 
+begin to run. In the above scenario, the `pod-4` can not be scheduled in time because a large number of small resource requirement
+pods occupied the node resources.
+ 
+In order to solve such problems, the reserve action is introduced into the scheduling policy. The `reserve` action can 
+prevents the large resource requirement task from being starved by reserving node resources for it and not allowing the other 
```

Comment:
How long will tasks be considered starved? Is it configured?

Reply:
This parameter can be configured, configured by the arguments parameter of the reserve action.